notifications/source/client.go

```diff
@@ -0,0 +1,187 @@
+package source
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"iter"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/icinga/icinga-go-library/notifications/event"
+	"github.com/pkg/errors"
+)
+
+// ErrRulesOutdated implies that the rules version between Icinga DB and Icinga Notifications mismatches.
+var ErrRulesOutdated = fmt.Errorf("rules version is outdated")
+
+// BasicAuthTransport is an http.RoundTripper that adds basic authentication and a User-Agent header to HTTP requests.
+type BasicAuthTransport struct {
```

Comment:
This type is only used internally in the implementation here, or this there actually a need to export it from the package?